newIDE/app/src/MainFrame/index.js

```diff
@@ -1889,13 +1889,23 @@ class MainFrame extends React.Component<Props, State> {
           (resourceSource, index): React.Node => {
             const Component = resourceSource.component;
             return (
-              <Component
-                key={resourceSource.name}
-                ref={dialog =>
-                  (this._resourceSourceDialogs[resourceSource.name] = dialog)
-                }
-                i18n={i18n}
-              />
+              <PreferencesContext.Consumer>
+                {({ loadLatestPath, saveLatestPath }) => {
+                  return (
+                    <Component
+                      key={resourceSource.name}
+                      ref={dialog =>
+                        (this._resourceSourceDialogs[
+                          resourceSource.name
+                        ] = dialog)
+                      }
+                      i18n={i18n}
+                      loadLatestPath={loadLatestPath}
+                      saveLatestPath={saveLatestPath}
```

Comment:
Name this function setLastUsedPath (or setLatestPath)

The fact that it's saving the preferences is "none of your business" at the level of this component. You just want the component to be able to say "here is the latest path for the project and this resource kind".
Then what you do, is up to you (you = the context).